flink-examples/flink-examples-streaming/pom.xml

```diff
@@ -60,6 +60,18 @@ under the License.
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-kafka-0.10_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
+			<exclusions>
```

Comment:
Why are you excluding this dependency here? end-to-end tests should be the final validation that our code works and having to modify dependencies doesn't sound good and might hide some error.

I wonder if maybe a better solution would be to create some separate sub modules for examples with kafka 0.10 and kafka 2.0?

@tzulitai what do you think?

Reply:
I also think that the Kafka 2.0 example should be in a separate module.
Take for example our Elasticsearch E2E tests: each version of the Elasticsearch job lives within its own module, so that we don't need to do any extra things with the dependency packaging.

On the other hand, as you can see, the Elasticsearch jobs live under the `flink-end-to-end-tests` module.
Perhaps it would make sense to do the same for Kafka, instead of relying on the released example jobs as we do right now?

Reply:
@pnowojski I ruled it out because I think the 2.0 version of the connector relies on the kafka client. But then I chose to exclude different connectors for different packaging modes. I can try it out and undo this code to see if it will cause problems.

Reply:
@pnowojski @tzulitai The result is not as expected (it failed), and even if we use `artifactSet / exculde` to exclude other connectors, the dependency on kafka-client is still wrong. So I'll take your advice and split the test implementation of the modern Kafka connector into separate modules. But because of the logical similarity, do you recommend that I introduce a base implementation (in a new base module)?

Reply:
@pnowojski I have split the modern kafka connector into a separate module, but I have not restored the code for the flink-examples-streaming module. Please review again (test failure is irrelevant).